src/OpenSpaceNet.cpp

```diff
@@ -614,26 +629,63 @@ void OpenSpaceNet::skipLine() const
     }
 }
 
-SizeSteps OpenSpaceNet::calcSizes() const
+cv::Size OpenSpaceNet::calcPrimaryWindowSize() const
 {
-    if(args_.pyramidWindowSizes.empty()) {
-        if(args_.pyramid) {
-            return SlidingWindow::calcSizes(bbox_.size(), model_->metadata().windowSize(), stepSize_, 2.0);
-        } else {
-            return { { model_->metadata().windowSize(), stepSize_ } };
-        }
-    } else {
-        // Sanity check, should've been caught before
-        DG_CHECK(args_.pyramidWindowSizes.size() == args_.pyramidStepSizes.size(), "Pyramid window sizes don't match step sizes.");
+    const auto& modelSize = model_->metadata().modelSize();
+    auto windowSize = model_->metadata().modelSize();
+    if(!args_.windowSizes.empty()) {
+        windowSize = {args_.windowSizes[0] * modelSize.height / modelSize.width, args_.windowSizes[0]};
```

Comment:
I thought width was going to stay constant, not height... Also, you have to cast the size to `float` or `double`, which you didn't do. It should look something like:
```cpp
windowSize = {args_.windowSizes[0], (int)roundf(args_.windowSizes[0]  * (float) modelSize.height / modelSize.width) };
```